gems/sorbet-runtime/lib/types/types/fixed_array.rb

```diff
@@ -19,8 +19,18 @@ def name
 
     # @override Base
     def valid?(obj)
-      obj.is_a?(Array) && obj.length == @types.length &&
-        obj.zip(@types).all? {|item, type| type.valid?(item)}
+      if obj.is_a?(Array) && obj.length == @types.length
+        i = 0
+        while i < @types.length
```

Comment:
Why an explicit `while` loop here? Will something like `@types.length.times` end up allocating?

Reply:
I'm not sure, I'll check!

Reply:
Looks like not:
```
irb(main):007:0> x = GC.stat[:total_allocated_objects]; GC.stat[:total_allocated_objects] - x - 1
=> 0
irb(main):008:0> x = GC.stat[:total_allocated_objects]; 2.times {}; GC.stat[:total_allocated_objects] - x - 1
=> 0
```

I can switch to `times` here if you prefer.

Reply:
That would be great, thank you!

Reply:
Ah, so this ends up producing an allocation if the type isn't valid because early-return from a block allocates. That probably doesn't matter because failing validation shouldn't be hot but I already wrote a test that covers this case so I'd rather just eat the slightly more complicated code here I think rather than unnecessarily allocate there. WDYT?

Reply:
That makes sense to me, thanks for taking a look!